### DISCUSSION TITLE
Possible Bug in interpolate_na

### DISCUSSION THREAD

**Discussion by Ockenfuss** [pydata/xarray#7547] [Q&A]
The following might be a bug in interpolate_na().
(If it is intended behaviour, I do not understand it).

Consider this array:
```python
import numpy as np
import xarray as xr
da=xr.DataArray([[np.nan,np.nan, np.nan, 2, np.nan, np.nan, 3], [1,np.nan, np.nan, 2, np.nan, np.nan, np.nan], [np.nan, np.nan, np.nan, 2, np.nan, np.nan, np.nan]], dims=["row", "column"])
da.plot(vmin=1, vmax=3)
```
Here is the corresponding plot: ![BeforeInterpolation](https://user-images.githubusercontent.com/42680748/220660424-5acece78-03ae-4395-8039-39dc172be36b.jpg)

Now, we interpolate the three rows along the "column" dimension:
```python
da=da.interpolate_na('column', 'nearest', limit=1, fill_value="extrapolate")
da.plot(vmin=1, vmax=3)
```
The results look surprising:  ![AfterInterpolation](https://user-images.githubusercontent.com/42680748/220660417-229ffaed-41c5-400e-a8f6-9cf171be1a6c.jpg)
For example, compare row 2 and 0: In row 0, column 0 a value of 2 was interpolated, which is not apparent in row 2. Those two rows differ only in the value of 3 at column 6, the opposite end of the array!
For me, it almost seems like periodic boundary conditions are applied, althoug this is not mentioned anywhere.
With linear interpolation, I get a similar behaviour. For linear interpolation, it is unclear which interpolation method is chosen internally, see issue #6899 .

Can anyone explain this behaviour or should I open an Issue?

Best,
Paul

**Comment by dcherian** [comment]
Is this fixed by your PR?

**Reply by Ockenfuss** [reply]
Not really, unfortunately. My PR (#7598) closes another issue I discovered accidentally while researching this one. I am still in the process of creating a proper example/bug report and will submit it hopefully next week. I will link it here as well.

**Comment by Ockenfuss** [comment]
I finally created an issue based on the above observations: #7665
